### PR TITLE
Streaming Pipeline

### PR DESCRIPTION
**Pull request by declark1**
Changes:

- Split unary and streaming handlers from `orchestrator.rs` to `unary.rs` and `streaming.rs`
- Rename `GenerationClient` enum to `GenerationClientInner` and wrap in `GenerationClient` unit struct to make it mockable with faux
- Implement `tokenize`, `generate`, and `generation_stream` methods on `GenerationClient`
- Refactor and rename some existing unary task handlers
- Implement streaming pipeline (see diagram below)
- Add `DetectionAggregator` trait and placeholder `MaxProcessedIndexAggregator` implementation
- Update `NlpClient`, `ChunkerClient`, and `TgisClient` streaming methods to return client streams directly rather than mpsc channels

Streaming Pipeline Diagram (DRAFT):
O8 Streaming Diagram

### CODE REVIEW

File: src/orchestrator/streaming.rs
Comment (gkumbhat) [inline src/orchestrator/streaming.rs:129]: nit: can we rename `chunk_streams` to `chunk_broadcast_streams` for readability

File: src/clients/nlp.rs
Comment (declark1) [reply src/clients/nlp.rs:91]: This is unrelated to that, it is just a simplification. Initially, I created a channel and wrapped it in a `ReceiverStream`. I am just returning the client response `Stream` directly now, i.e. the channel is unnecessary.

```rust
let (tx, rx) = mpsc::channel(128);
tokio::spawn(async move {
    while let Some(Ok(message)) = response_stream.next().await {
        let _ = tx.send(message).await;
    }
});
Ok(ReceiverStream::new(rx))
```

File: src/orchestrator/processors/max_processed_index.rs
Comment (declark1) [reply src/orchestrator/processors/max_processed_index.rs]: Renamed trait to `DetectionAggregator` (should probably at least have "detection" in the name) and `MaxProcessedIndexAggregator`. Let me know if you are good with this.

File: src/orchestrator/streaming.rs
Comment (gkumbhat) [inline src/orchestrator/streaming.rs:92]: in case `output_detectors` are empty, it would still be `Some({})` object, so it will get matched here and go into this `if` condition, right? However, we would want it to go to `else` so that we can just process `generation_stream` ? 

Essentially, looks like we may not be following the generation when no detections are requested case ?

Comment (declark1) [reply src/orchestrator/streaming.rs:92]: This is really an incorrect use of `Option`, i.e. it should really be `None` rather than `Some({})` when there are no detectors. I will update it to handle this properly, since it appears we want to do the following rather than `"output": null` or simply omitting the `output` field?

```json
"guardrail_config": {
  "output": {
     "models": {}
  }
}
```

It would probably be good to clean this up during deserialization rather than handling empty nested objects.

Comment (declark1) [reply src/orchestrator/streaming.rs]: Pushed fix for this

File: src/models.rs
Comment (declark1) [reply src/models.rs:694]: I don't believe this is feasible as we are returning a different response object. The TGIS API is stable at this point anyway, the only recent change was to add a new field to specify LoRA/other adapters.

**Review by gkumbhat — COMMENTED** [review]
Thanks Dan for all the changes.  There are some conflicts remaining, probably arising from rebasing that we need to address, but other than that the overall changes LGTM.

**Comment by declark1** [comment]
> Thanks Dan for all the changes. There are some conflicts remaining, probably arising from rebasing that we need to address, but other than that the overall changes LGTM.

Thanks. It should be good to merge now once you approve.

**Review by gkumbhat — APPROVED** [review]
Looks good to me!

**Pull request merged**
